ActionScript/ActionScript.sublime-syntax

```diff
@@ -12,87 +12,892 @@ first_line_match: |-
+      pop: 1
+
+    - match: ^\s*(\*)(?!/)
       captures:
-        1: keyword.declaration.class.actionscript.2
-        2: entity.name.type.class.actionscript.2
-        3: storage.modifier.extends.actionscript.2
-        4: entity.other.inherited-class.actionscript.2
+        1: punctuation.definition.comment.actionscript
+
+  line-comments:
+    - include: line-comment-double-slash
+
+  line-comment-double-slash:
+    - match: //
+      scope: punctuation.definition.comment.actionscript
+      push: line-comment-double-slash-content
+
+  line-comment-double-slash-content:
+    - meta_include_prototype: false
+    - meta_scope: comment.line.double-slash.actionscript
+
+    - include: eol-pop
+
+##[ Strings ]###########################################################################################################
+
+  strings:
+    - include: double-quoted-strings
+    - include: single-quoted-strings
+
+  double-quoted-strings:
+    - match: \"
+      scope: punctuation.definition.string.begin.actionscript
+      push: double-quoted-string-content
+
+  double-quoted-string-content:
+    - meta_include_prototype: false
+    - meta_scope: string.quoted.double.actionscript
+
+    - match: \"
+      scope: punctuation.definition.string.end.actionscript
+      pop: 1
+
+    - match: \\.
+      scope: constant.character.escape.actionscript
+
+  single-quoted-strings:
+    - match: \'
+      scope: punctuation.definition.string.begin.actionscript
+      push: single-quoted-string-content
+
+  single-quoted-string-content:
+    - meta_include_prototype: false
+    - meta_scope: string.quoted.single.actionscript
+
+    - match: \'
+      scope: punctuation.definition.string.end.actionscript
+      pop: 1
+
+    - match: \\.
+      scope: constant.character.escape.actionscript
+
+##[ Constants ]#########################################################################################################
+
+  constants:
+    - include: language-constants
+    - include: numeric-constants
+    - include: keyboard-input-constants
+
+  language-constants:
+    - match: \b{{constant_language}}\b
+      scope: constant.language.actionscript
+
+  numeric-constants:
+    - match: ([-+]??)(0[xX])(\h*)(L|l|UL|ul|u|U|F|f)?
+      scope: meta.number.integer.hexadecimal.actionscript
+      captures:
+        1: constant.numeric.sign.actionscript
+        2: constant.numeric.base.actionscript
+        3: constant.numeric.value.actionscript
+        4: constant.numeric.suffix.actionscript
+
+    - match: |-
+        (?x:
+          (\d+\.?\d*|\.\d+)
+          ({{constant_numeric_exponent}})?
+          (L|l|UL|ul|u|U|F|f)?
+        )
+      scope: meta.number.actionscript
+      captures:
+        1: constant.numeric.value.actionscript
+        2: constant.numeric.actionscript
+        3: constant.numeric.suffix.actionscript
+
+  keyboard-input-constants:
+    - match: \b{{constant_keyboard_inputs}}\b
+      scope: support.constant.actionscript
```

Comment:
There's another  `\b(?:{{constants}})\b` pattern in `main` context. I'd suggest to create a dedicated `language-constants` context to group all kinds of known/predefined constants.